type-generator/src/main/java/io/mesosphere/types/dtr/compiler/internal/CompilerEngine.java

```diff
@@ -0,0 +1,46 @@
+package io.mesosphere.types.dtr.compiler.internal;
+
+import io.mesosphere.types.dtr.models.internal.Type;
+import io.mesosphere.types.dtr.models.internal.TypeScope;
+import io.mesosphere.types.dtr.models.internal.Value;
+
+/**
+ * A compiler engine is the logic that converts a template name + macro values into
+ * the final string, according to the engine specifications.
+ */
+public interface CompilerEngine {
+
+    /**
+     * Return the implementation-dependent name of the storage type
+     * </p>
+     * For example `StringValue` would be `String` in Java
+     * </p>
+     * Note that array types or other more complicated structures are constructed
+     * through the engine templates and we don't need to provide such functionality here.
+     *
+     * @param storageType The internal storage type for which to get a name
+     * @return Returns a string with the name of the storage type in the user-defined implementation
+     */
+    String getStorageTypeName(Value storageType);
```

Comment:
Hm, `StorageType` is an odd name. Could this be done in StringTemplate? I mean the internal model would know which type it is so the template could just decide how to name it.

Reply:
As a matter of fact yes. This could be removed. This is a remainder of the initial moustache implementation ...